src/AutoBuild.jl

```diff
@@ -1171,8 +1171,25 @@ function rebuild_jll_package(name::String, build_version::VersionNumber, sources
     build_output_meta = Dict()
 
     # For each platform, we have two tarballs: the main with the full product,
-    # and the logs-only one.  This function filters out the logs one.
-    filter_main_tarball(f, platform) = occursin(".$(triplet(platform)).tar", f) && !occursin("-logs.", f)
+    # and the logs-only one.  This function filters out the logs one, and
+    # finds the tarball matching `platform`.
+    function filter_main_tarball(tarball_filename, platform)
+        if occursin("-logs.", tarball_filename)
+            return false
+        end
+        tarball_filename_match = match(r"^(?<name>[\w_]+)\.v(?<version>\d+\.\d+\.\d+)\.(?<platform_triplet>([^-]+-?)+).tar", tarball_filename)
+        if isnothing(tarball_filename_match)
+            @warn "Tarball filename does not match expected pattern: $(tarball_filename)"
+            return false
+        end
+        try
+            tarball_filename_platform = parse(Platform, tarball_filename_match[:platform_triplet])
+            return tarball_filename_platform == platform
```

Comment:
One thing I'm not 100% sure is whether this comparison is fine. Platform equality is finicky, I usually prefer using platform_match, but probably in this case we _really_ want to check exact equality specifically, so this may be good.

Reply:
Good points - both of them. I’ll check up on `platform_match`.

Reply:
```
julia> using Base.BinaryPlatforms

julia> p1 = Platform("x86_64", "linux"; tag1="foo", tag3="blah")
Linux x86_64 {libc=glibc, tag1=foo, tag3=blah}

julia> p2 = Platform("x86_64", "linux"; tag2="bar", tag4="glah")
Linux x86_64 {libc=glibc, tag2=bar, tag4=glah}

julia> platforms_match(p1, p2)
true

julia> p3 = Platform("x86_64", "linux"; tag3="blah", tag1="foo")
Linux x86_64 {libc=glibc, tag1=foo, tag3=blah}

julia> p4 = Platform("x86_64", "linux"; tag4="glah", tag2="bar")
Linux x86_64 {libc=glibc, tag2=bar, tag4=glah}

julia> platforms_match(p1, p3)
true

julia> p1 == p3
true

julia> p2 == p4
true
```
Probably equality is better than platform_match here, but make sure with appropriate tests that it works correctly also in complex cases.

Reply:
I have added some tests.

I would argue that the functionality of the `Platform` constructor should not be tested here (p1 vs p3, and p2 vs p4), but the difference between `Platform` equality and `platforms_match` should, e.g. where `platforms_match` is true and equality is false,
```
julia> p5 = Platform("x86_64", "linux")
Linux x86_64 {libc=glibc}

julia> p6 = Platform("x86_64", "linux"; cxxstring_abi="cxx11")
Linux x86_64 {cxxstring_abi=cxx11, libc=glibc}

julia> platforms_match(p5, p6)
true

julia> p5 == p6
false
```